src/main/java/com/product/dao/ProductDao.java

```diff
@@ -0,0 +1,20 @@
+package com.product.dao;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestTemplate;
+
+import com.product.model.Customer;
+import com.product.model.Product;
+
+@Repository
+public interface ProductDao extends MongoRepository<Product, Integer> {
```

Comment:
You need to create `ProductDao` and `ProductDaoImpl` class, as you did with `CustomerDao` and CustomerDaoImpl.

The interface should never have declarations like your . class has. Only the Impl class should have the logic, while the interface . should just define the Skeleton of the methods.

Reply:
Done.